### PR TITLE
Set up Documentation

### PR DESCRIPTION
**Pull request by bipol**
This PR leverages [swag](https://github.com/swaggo/swag) to generate swagger documentation given a set of annotations written into our handlers.

Added a  `make docs` to generate documentation after adding annotations to your handlers.

This PR pulled in a ton of deps, which we see in our files changes (and they all live in vendor).

image

### CODE REVIEW

File: pkg/controllers/live.go [hunk 1]
```diff
@@ -2,18 +2,27 @@ package controllers
...
 	"github.com/smartatransit/third_rail/pkg/validators"
-	"log"
-	"net/http"
 )
 
 type LiveController struct {
 	MartaClient clients.MartaClient
 }
 
+// GetScheduleByLine godoc
```
Comment (bipol) [inline pkg/controllers/live.go:18]: These are the annotations that describe the documentation -- I may have missed some parameters or gave a poor description

File: pkg/controllers/smart.go [hunk 2]
```diff
@@ -2,15 +2,23 @@ package controllers
...
 	TwitterClient clients.TwitterClient
 }
 
+// GetParkingStatus godoc
+// @Summary Get Parking Information
+// @Description  Get available parking information as informed by twitter
+// @Produce  json
+// @Success 200 {object} parkingResponse
+// @Router /smart/parking [get]
+// @Security ApiKeyAuth
```
Comment (bipol) [inline pkg/controllers/smart.go:21]: I noticed that some of the other endpoints on this file aren't exposed yet, so I didn't annotate them

**Review by xanderflood — APPROVED** [review]
This looks awesome! Is there any plan to deploy the API docs somewhere? Would be cool to see if we can deploy these to a lil nginx container via CI

**Comment by xanderflood** [comment]
(one approach would be to have this repo build and push `third-rail-docs` image in addition to the service image)

**Comment by bipol** [comment]
@xanderflood so, the docs will be exposed off of the endpoint `/swagger/index.html#/`(e.g. https://third-rail.services.ataper.net/swagger/index.html) - so we should get that for free on deploy.  Definitely did consider doing some more `build-fu` in the future though!

**Review by trillnye — APPROVED** [review]
absolutely fucking awesome - great work, dude!

**Pull request merged**